src/conn/opts.rs

```diff
@@ -461,6 +461,43 @@ impl OptsBuilder {
         OptsBuilder { opts: opts.into() }
     }
 
+    /// Use a HashMap for creating creating an OptsBuilder instance:
+    /// ```ignore
+    /// OptsBuilder::new().from_has_map(client);
+    /// ```
+    /// `HashMap` key,value pairs:
+    /// - user = username
+    /// - password = password
+    /// - host = host name or ip address
+    /// - port = port, default is 3306
+    /// - socket = unix socket or pipe name(on windows) defaults to `None`
+    ///
+    /// Login .cnf file parsing lib https://github.com/rjcortese/myloginrs returns a HashMap for client configs
+    ///
+    /// **Note:** You do **not** have to use myloginrs lib.    
+    pub fn from_hash_map(mut self, client: &HashMap<String, String>) -> Self {
```

Comment:
I think it'd be reasonable to either:
* rename it to `from_mylogin_cnf` because the list of supported options is limited by the `mysql_config_editor` capabilities;
* or to add support for [other url parameters](https://docs.rs/mysql/20.0.1/mysql/#url-based-connection-string).

I'd personally suggest the second option.

Reply:
I'll work to add the second option